From 5f066c4125d7a648958b31b909934e02cf397db8 Mon Sep 17 00:00:00 2001
From: Jiri Konecny <jkonecny@redhat.com>
Date: Mon, 8 Feb 2021 11:27:26 +0100
Subject: [PATCH 2/3] Revert "Do not support no inst. Anaconda boot args in
 systemd services (#1912502)" (#1907566)

Allow to use Anaconda kernel boot arguments systemd units to not break beaker.
Beaker needs more time to adaptation so avoid to break it for RHEL-9.

This reverts commit 322e6a847d73ea17fe99aaae4129e14409488b6b.
Related: rhbz#1907566
---
 data/systemd/anaconda-noshell.service | 1 +
 data/systemd/anaconda-shell@.service  | 1 +
 data/systemd/anaconda-sshd.service    | 2 ++
 3 files changed, 4 insertions(+)

diff --git a/data/systemd/anaconda-noshell.service b/data/systemd/anaconda-noshell.service
index ca4dbd328d..f378c67b0c 100644
--- a/data/systemd/anaconda-noshell.service
+++ b/data/systemd/anaconda-noshell.service
@@ -3,6 +3,7 @@ Description=Restrict Anaconda Text Console
 After=anaconda.service
 Before=anaconda-tmux@.service
 ConditionKernelCommandLine=|inst.noshell
+ConditionKernelCommandLine=|noshell
 
 [Service]
 Type=oneshot
diff --git a/data/systemd/anaconda-shell@.service b/data/systemd/anaconda-shell@.service
index b0dd4b7c66..bc69fd1208 100644
--- a/data/systemd/anaconda-shell@.service
+++ b/data/systemd/anaconda-shell@.service
@@ -3,6 +3,7 @@
 [Unit]
 Description=Shell on %I
 After=systemd-user-sessions.service plymouth-quit-wait.service
+ConditionKernelCommandLine=!noshell
 ConditionKernelCommandLine=!inst.noshell
 
 [Service]
diff --git a/data/systemd/anaconda-sshd.service b/data/systemd/anaconda-sshd.service
index 8b5b16fbbe..8cba2c4bb0 100644
--- a/data/systemd/anaconda-sshd.service
+++ b/data/systemd/anaconda-sshd.service
@@ -3,8 +3,10 @@ Description=OpenSSH server daemon
 Before=anaconda.target
 After=syslog.target network.target sshd-keygen.target
 Wants=sshd-keygen.target
+ConditionKernelCommandLine=|sshd
 ConditionKernelCommandLine=|inst.sshd
 ConditionKernelCommandLine=!inst.sshd=0
+ConditionKernelCommandLine=!sshd=0
 # TODO: use ConditionArchitecture in systemd v210 or later
 ConditionPathIsDirectory=|/sys/hypervisor/s390
 
-- 
2.29.2

